### ISSUE TITLE
Python 3.10 fails when __index__ is not implemented

### ISSUE THREAD

**Issue by Astranox** [performancecopilot/pcp#1410]
Running dstat with python 3.10 results in this error message:
```
You did not select any stats, using -cdngy by default.
Traceback (most recent call last):
  File "/usr/bin/dstat", line 1825, in <module>
    dstat.execute()
  File "/usr/bin/dstat", line 1799, in execute
    self.context.prepare_execute(self.opts, False, self.interpol, self.interval)
  File "/usr/lib64/python3.10/site-packages/pcp/pmapi.py", line 2640, in prepare_execute
    self.set_timezone(options)
  File "/usr/lib64/python3.10/site-packages/pcp/pmapi.py", line 2596, in set_timezone
    timezone = pmContext.get_current_tz(options)
  File "/usr/lib64/python3.10/site-packages/pcp/pmapi.py", line 2561, in get_current_tz
    dst = time.localtime(options.pmGetOptionOrigin()).tm_isdst
TypeError: 'timeval' object cannot be interpreted as an integer
```

This seems to have its source in the class `timeval` [1] not implementing `__index__`, which seems to be mandatory [2], probably due to python3.10 dropping python2 quirks (python 3.9.7 docs mention: "Presence of this method indicates that the numeric object is an integer type.").

[1] https://github.com/performancecopilot/pcp/blob/25e0b92debc842a6cc9e53057777296827f4eb90/src/python/pcp/pmapi.py#L227
[2] https://docs.python.org/release/3.9.7/reference/datamodel.html#object.__index__

**Comment by natoscott** [comment]
@Astranox thanks for the report and diagnosis.  Looks like we've arrived at the same place independently -  pcp-5.3.3 (releasing today) will contain this commit...

```
commit a22c5d3275a871463121ba517d82f5855eeb84cf
Author: Nathan Scott <nathans@redhat.com>
Date:   Tue Aug 24 14:14:27 2021 +1000

    python api: additional converter functions for struct timeval
    
    This makes pcp-dstat function with python 3.10 in rawhide.
    
    Resolves Fedora BZ 1996324.
```

**Comment by Astranox** [comment]
Ah, thanks, didn't see the fedora bug report, only searched for NEW and MODIFIED, not POST. Sorry for the noise.
